driver-sync/src/test/functional/com/mongodb/client/unified/ValueMatcher.java

```diff
@@ -39,92 +41,105 @@ final class ValueMatcher {
+                                break;
+                            default:
+                                throw new UnsupportedOperationException("Unsupported special operator: " + value.asDocument().getFirstKey());
+                        }
                     }
-                }
 
-                assertTrue(context.getMessage("Actual document must contain key " + key), actualDocument.containsKey(key));
-                assertValuesMatch(value, actualDocument.get(key), false, key, -1);
-            });
-        } else if (expected.isArray()) {
-            assertTrue(context.getMessage("Actual value must be an array but is " + actual.getBsonType()), actual.isArray());
-            assertEquals(context.getMessage("Arrays must be the same size"), expected.asArray().size(), actual.asArray().size());
-            for (int i = 0; i < expected.asArray().size(); i++) {
-                assertValuesMatch(expected.asArray().get(i), actual.asArray().get(i), false, keyContext, i);
+                    assertTrue(context.getMessage("Actual document must contain key " + key), actualDocument.containsKey(key));
+                    assertValuesMatch(value, actualDocument.get(key), key, -1);
+                });
+            } else if (expected.isArray()) {
+                assertTrue(context.getMessage("Actual value must be an array but is " + actual.getBsonType()), actual.isArray());
+                assertEquals(context.getMessage("Arrays must be the same size"), expected.asArray().size(), actual.asArray().size());
+                for (int i = 0; i < expected.asArray().size(); i++) {
+                    assertValuesMatch(expected.asArray().get(i), actual.asArray().get(i), keyContext, i);
+                }
+            } else if (expected.isNumber()) {
+                assertTrue(context.getMessage("Expected a number"), actual.isNumber());
+                assertEquals(context.getMessage("Expected BSON numbers to be equal"),
+                        expected.asNumber().doubleValue(), actual.asNumber().doubleValue(), 0.0);
+            } else {
+                assertEquals(context.getMessage("Expected BSON types to be equal"), expected.getBsonType(), actual.getBsonType());
+                assertEquals(context.getMessage("Expected BSON values to be equal"), expected, actual);
             }
-        } else if (expected.isNumber()) {
-            assertTrue(context.getMessage("Expected a number"), actual.isNumber());
-            assertEquals(context.getMessage("Expected BSON numbers to be equal"),
-                    expected.asNumber().doubleValue(), actual.asNumber().doubleValue(), 0.0);
-        } else {
-            assertEquals(context.getMessage("Expected BSON types to be equal"), expected.getBsonType(), actual.getBsonType());
-            assertEquals(context.getMessage("Expected BSON values to be equal"), expected, actual);
+        } finally {
```

Comment:
Pop in the finally block since there are multiple early returns from this method.